compiler/src/dmd/escape.d

```diff
@@ -1254,6 +1256,15 @@ private bool checkReturnEscapeImpl(ref Scope sc, Expression e, bool refs, bool g
                             "perhaps annotate the %s with `return`", annotateKind);
                     }
                 }
+                else if (sc.func.isTrusted())
```

Comment:
As mentioned [on bugzilla](https://issues.dlang.org/show_bug.cgi?id=24724#c1), there shouldn't be special cases for `@trusted` compared to `@system` with respect to restrictions in the function body.